src/type-exercises.ts

```diff
@@ -0,0 +1,262 @@
+/*eslint-disable*/
+
+
+/**
+ * Exercise #1: Filter object properties by type.
+ *
+ * Using a utility type `OmitByType`, this example demonstrates how to pick properties
+ * from a type `T` whose values are *not* assignable to a specified type `U`.
+ *
+ * @example
+ * type OmitBoolean = OmitByType<{
+ *   name: string;
+ *   count: number;
+ *   isReadonly: boolean;
+ *   isEnable: boolean;
+ * }, boolean>;
+ *
+ * Resulting type:
+ *
+ * {
+ * name: string;
+ * count: number;
+ * }
+ */
+
+
+// Add here your solution
+const greeting = <T, U>(toFilter: T, toOmmit: U) => {
+    let finalObject: Partial<T> = {};
+    for (const key in toFilter) {
+        if (typeof toFilter[key] !== typeof toOmmit) {
+            finalObject[key] = toFilter[key];
+        }
+    }
+    return finalObject as Partial<T>;
+}
+
+interface customType {
+    name: string,
+    count: number,
+    isReadonly: boolean,
+    isEnable: boolean
+}
+// Add here your example
+console.log('Exercise #1: Filter object properties by type ✨');
+console.log('Ommiting string type from the object');
+console.log(
+    greeting<customType, string>({
+        name: 'test',
+        count: 1,
+        isReadonly: true,
+        isEnable: false
+    }, '')
+);
+
+console.log('Ommiting boolean type from the object');
+console.log(
+    greeting<customType, boolean>({
+        name: 'test',
+        count: 1,
+        isReadonly: true,
+        isEnable: false
+    }, false)
+);
+
+console.log('Ommiting number type from the object');
+console.log(
+    greeting<customType, number>({
+        name: 'test',
+        count: 1,
+        isReadonly: true,
+        isEnable: false
+    }, 0)
+);
+
+/**
+ * Exercise #2: Implement the utility type `If<C, T, F>`, which evaluates a condition `C`
+ * and returns one of two possible types:
+ * - `T` if `C` is `true`
+ * - `F` if `C` is `false`
+ *
+ * @description
+ * - `C` is expected to be either `true` or `false`.
+ * - `T` and `F` can be any type.
+ *
+ * @example
+ * type A = If<true, 'a', 'b'>;  // expected to be 'a'
+ * type B = If<false, 'a', 'b'>; // expected to be 'b'
+ */
+
+// Add here your solution
+type If<C extends boolean, T, F> = C extends true ? T : F;
```

Comment:
All good here. Just remember that we can use more elaborated names for the generics too.
```typescript
type If<Condition extends boolean, TrueValue, FalseValue> = Condition extends true ? TrueValue : FalseValue;
// or 
type If<TCondition extends boolean, TTrueValue, TFalseValue> = TCondition extends true ? TTrueValue : TFalseValue;
```